internal/resources/cluster/resource_cluster.go

```diff
@@ -563,6 +593,57 @@ func withMetaUpdate(d *schema.ResourceData, cluster *clustermodel.VmwareTanzuMan
 	return true
 }
 
+func withTKGNodePoolUpdate(d *schema.ResourceData, nodepool *nodepoolmodel.VmwareTanzuManageV1alpha1ClusterNodepoolNodepool) bool {
+	switch {
+	case nodepool.Spec.TkgServiceVsphere != nil:
+		nodepools := d.Get(helper.GetFirstElementOf(SpecKey, tkgServiceVsphereKey, topologyKey, nodePoolKey)).([]interface{})[0].(map[string]interface{})
+		poolSpec := nodepools[SpecKey].([]interface{})[0].(map[string]interface{})
+		// poolInfo := nodepools["info"].([]interface{})[0].(map[string]interface{})
+		tkgsSpec := poolSpec[tkgServiceVsphereKey].([]interface{})[0].(map[string]interface{})
+
+		if d.HasChange(helper.GetFirstElementOf(SpecKey, tkgServiceVsphereKey, topologyKey, nodePoolKey)) {
+			incomingWorkerNodeCount := poolSpec[workerNodeCountKey].(string)
+
+			if incomingWorkerNodeCount != "" {
+				nodepool.Spec.WorkerNodeCount = incomingWorkerNodeCount
+			}
+
+			incomingTkgServiceVsphereClass := tkgsSpec[classKey].(string)
+
+			if incomingTkgServiceVsphereClass != "" {
+				nodepool.Spec.TkgServiceVsphere.Class = incomingTkgServiceVsphereClass
+			}
+
+			incomingTkgServiceVsphereStorageClass := tkgsSpec[storageClassKey].(string)
+
+			if incomingTkgServiceVsphereStorageClass != "" {
+				nodepool.Spec.TkgServiceVsphere.StorageClass = incomingTkgServiceVsphereStorageClass
+			}
+
+			log.Printf("[INFO] updating TKGs workload cluster nodepools")
+
+			return true
+		}
+	case nodepool.Spec.TkgVsphere != nil:
+		nodepools := d.Get(helper.GetFirstElementOf(SpecKey, tkgVsphereClusterKey, topologyKey, nodePoolKey)).([]interface{})[0].(map[string]interface{})
+		poolSpec := nodepools[SpecKey].([]interface{})[0].(map[string]interface{})
+
+		if d.HasChange(helper.GetFirstElementOf(SpecKey, tkgVsphereClusterKey, topologyKey, nodePoolKey)) {
+			incomingWorkerNodeCount := poolSpec[workerNodeCountKey].(string)
+
+			if incomingWorkerNodeCount != "" {
+				nodepool.Spec.WorkerNodeCount = incomingWorkerNodeCount
```

Comment:
only worked node count is updatable in TKGvSphere nodepool?

Reply:
this is what the current implementation allows through the nodepools resource. I did not change this functionality for this PR. We could add more updateable fields but I think that should be a second PR so we can get this feature delivered for the customer ASAP. see current implementation here https://github.com/vmware/terraform-provider-tanzu-mission-control/blob/main/internal/resources/cluster/nodepools/resource_node_pool.go#L646-L660